devicetypes/smartthings/orvibo-gas-detector.src/Orvibo-Gas-detector.groovy

```diff
@@ -0,0 +1,123 @@
+			state "strobe", label:'strobe!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+			state "siren", label:'siren!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+			state "both", label:'alarm!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		main "main"
+		details(["main","alarm","refresh"])
+	}
+}
+def installed(){
+	log.debug "installed"
+	return zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER,zigbee.ATTRIBUTE_IAS_ZONE_STATUS) +
+					zigbee.readAttribute(0x0009,0x0000)
+}
+def parse(String description) {
+	log.debug "description(): $description"
+	def map = zigbee.getEvent(description)
+	if(!map){
+		if (description?.startsWith('zone status')) {
+			map = parseIasMessage(description)
+		}
+	}
+	log.debug "Parse returned $map"
+	def result = map ? createEvent(map) : [:]
+	if (description?.startsWith('enroll request')) {
+		List cmds = zigbee.enrollResponse()
+		log.debug "enroll response: ${cmds}"
+		result = cmds?.collect { new physicalgraph.device.HubAction(it)}
+	}
+	return result
+}
+private Map parseIasMessage(String description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+	translateZoneStatus(zs)
+}
+private Map translateZoneStatus(ZoneStatus zs) {
+	// Some sensor models that use this DTH use alarm1 and some use alarm2 to signify motion
+	return (zs.isAlarm1Set() || zs.isAlarm2Set()) ? getDetectedResult('detected') : getDetectedResult('clear')
+}
+private Map getDetectedResult(value) {
+	String descriptionText = value == 'detected' ? "${device.displayName} detected smoke" : "${device.displayName} smoke clear"
+	if(value == detected){
+		strobe()
+	}else{
+		off()
+	}
+	return [
+			name			: 'smoke',
+			value			: value,
+			descriptionText : descriptionText,
+			translatable	: true
+	]
+}
+def strobe() {
+	sendEvent(name: "alarm", value: "strobe")
+}
+def siren() {
+	sendEvent(name: "alarm", value: "siren")
+}
+def both() {
+	sendEvent(name: "alarm", value: "both")
+}
+def off() {
+	sendEvent(name: "alarm", value: "off")
+}
+def refresh() {
+	log.debug "Refreshing Values"
+	def refreshCmds = []
+	refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021) +
+					zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER,zigbee.ATTRIBUTE_IAS_ZONE_STATUS) +
+					zigbee.enrollResponse()
+	return refreshCmds
+}
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	log.debug "ping "
+	return zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS)
+}
+def configure() {
+	log.debug "configure"
+	sendEvent(name: "checkInterval", value:40 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```

Comment:
Due to possibility of network lag add some additional timeout to this like 2 mins